### ISSUE TITLE
`CALL DOLT_CHECKOUT()` on an existing branch give "branch already exists" error

### ISSUE THREAD

**Issue by timsehn** [dolthub/dolt#4290]
```
mysql> CALL DOLT_CHECKOUT('8_0_1');
ERROR 1105 (HY000): fatal: A branch named '8_0_1' already exists.
```

Obviously, this should just switch to the branch if it already exists.

**Comment by timsehn** [comment]
Not a simple repro.
```
mysql> select * from dolt_branches;
+--------+----------------------------------+------------------+------------------------+-------------------------+-----------------------+
| name   | hash                             | latest_committer | latest_committer_email | latest_commit_date      | latest_commit_message |
+--------+----------------------------------+------------------+------------------------+-------------------------+-----------------------+
| branch | cd4fkf36jlpbg4p84kofkt63unk1tt5f | Tim Sehn         | tim@dolthub.com        | 2022-09-01 20:18:59.754 | created table         |
| main   | cd4fkf36jlpbg4p84kofkt63unk1tt5f | Tim Sehn         | tim@dolthub.com        | 2022-09-01 20:18:59.754 | created table         |
+--------+----------------------------------+------------------+------------------------+-------------------------+-----------------------+
2 rows in set (0.00 sec)

mysql> call dolt_checkout('branch');
+--------+
| status |
+--------+
|      0 |
+--------+
1 row in set (0.00 sec)

mysql> call dolt_checkout('main');
+--------+
| status |
+--------+
|      0 |
+--------+
1 row in set (0.00 sec)

mysql>
```

**Comment by timsehn** [comment]
Not underscores:
```
mysql> call dolt_checkout('-b', '8_0_1');
+--------+
| status |
+--------+
|      0 |
+--------+
1 row in set (0.00 sec)

mysql> call dolt_checkout('main');
+--------+
| status |
+--------+
|      0 |
+--------+
1 row in set (0.00 sec)

mysql> call dolt_checkout('8_0_1');
+--------+
| status |
+--------+
|      0 |
+--------+
1 row in set (0.00 sec)
```

**Comment by timsehn** [comment]
More info:
```
mysql> CALL DOLT_CHECKOUT('main');
+--------+
| status |
+--------+
|      0 |
+--------+
1 row in set (0,08 sec)

mysql> CALL DOLT_CHECKOUT('8_0_1');
ERROR 1105 (HY000): fatal: A branch named '8_0_1' already exists.
mysql> CALL DOLT_CHECKOUT('35-new-branch-1');
+--------+
| status |
+--------+
|      0 |
+--------+
1 row in set (0,07 sec)

mysql> CALL DOLT_CHECKOUT('incompatible_genes_to_new_mutation_type');
ERROR 1105 (HY000): fatal: A branch named 'incompatible_genes_to_new_mutation_type' already exists.
mysql> CALL DOLT_CHECKOUT('netex-20220302-151951-gkovacs-mk3vcoqx65');
ERROR 1105 (HY000): fatal: A branch named 'netex-20220302-151951-gkovacs-mk3vcoqx65' already exists.
mysql> 
```

**Comment by timsehn** [comment]
Some branches work, like `35-new-branch-1`.

**Comment by timsehn** [comment]
@reltuk fixed this in #4300
